include/arkode/arkode_arkstep.h

```diff
@@ -403,6 +405,14 @@ SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodeFree instead")
 void ARKStepFree(void** arkode_mem);
 SUNDIALS_DEPRECATED_EXPORT_MSG("use ARKodePrintMem instead")
 void ARKStepPrintMem(void* arkode_mem, FILE* outfile);
+
+/* Adjoint solver functions */
+SUNDIALS_EXPORT
+int ARKStepCreateAdjointStepper(void* arkode_mem, SUNAdjRhsFn adj_fe,
+                                SUNAdjRhsFn adj_fi, sunrealtype tf, N_Vector sf,
+                                SUNAdjointStepper* adj_stepper_ptr);
```

Comment:
Should we pass in the context explicitly to the constructor?

Reply:
I dont think we should because it should probably use the same one as the original arkode_mem.

Reply:
Well, im not fully sure that using separate SUNContext would be advisable, its probably better to add it as an arg for consistency and so we do not need a major API change to support it. If it does turn out to be an issue, we can add a check that the two contexts are the same.